lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -283,7 +285,10 @@ sub job_grab {
         $job_hashref->{settings}->{NICVLAN} = join(',', @vlans);
     }
 
-    # TODO: cleanup previous tmpdir
+    if (-e $worker->get_property('WORKER_TMPDIR')) {
```

Comment:
We colud add here a check if the previous job results were uploaded (at least the autoinst.txt log file...)... and then carry on